pkg/segment/utils/segconsts.go

```diff
@@ -1129,14 +1129,12 @@ func (e *CValueEnclosure) WriteToBytesWithType(buf []byte, bufIdx int) ([]byte,
 	case SS_DT_SIGNED_NUM:
 		copy(buf[bufIdx:], VALTYPE_ENC_INT64)
 		bufIdx += 1
-		bytesVal := toputils.Int64ToBytesLittleEndian(e.CVal.(int64))
-		copy(buf[bufIdx:], bytesVal)
+		toputils.Int64ToBytesLittleEndianInplace(e.CVal.(int64), buf[bufIdx:bufIdx+8])
```

Comment:
You did not do the same for `uint64`. It is still doing `Uint64ToBytesLittleEndian`.